### PR TITLE
added list.swap()

### PR DESCRIPTION
**Pull request by HosseinYousefi**
It's quite useful to swap two elements in a list. For example in flutter's `ReorderableListView`. Here's a very simple implementation of it.

### CODE REVIEW

File: lib/src/list.dart [hunk 1]
```diff
@@ -153,6 +153,18 @@ extension ListX<E> on List<E> {
...
     _mergeSort(this, start: start, end: end, compare: comparator);
   }
+
+  /// Swaps the elements in the indices provided.
+  ///
+  /// ```dart
+  /// var list = [1, 2, 3, 4];
+  /// list.swap(0, 2); // [3, 2, 1, 4]
+  /// ```
+  void swap(int i, int j) {
```
Comment (passsy) [inline lib/src/list.dart:163]: Please make this function generic and use `T` instead of `int`.

**Comment by passsy** [comment]
Please also add tests when accessing out-of-bounds indices

File: lib/src/list.dart (hunk 1 above)
Comment (HosseinYousefi) [reply lib/src/list.dart:163]: like this: `void swap<T extends int>(T i, T j)`? since they are indices afterall and should be `int`s.

**Comment by HosseinYousefi** [comment]
Added the test for out-of-bound indices. wanted your clarification on making the function generic.

File: lib/src/list.dart (hunk 1 above)
Comment (passsy) [reply lib/src/list.dart:163]: My dumb brain mad a mistake. `int` is 100% correct.

**Pull request merged**